### ISSUE TITLE
/sync crashes on normal youtube video

### ISSUE THREAD

**Issue by chauchausoup** [0chain/0proxy#39]
May be 0proxy is not properly handled to work with normal youtube videos.
While checking the 0proxy logs on the container its says the file is downloaded 100% status  info is shown, on performing GET /sync is gives proxy is busy. Issue is that container crashes and uses maximum CPU and only way to recover 0proxy is by restarting it.
We need to handle this behaviour and prevent crashing or 0proxy from malicious or crashful urls. by checking at first if its a live steam or not. One way I did it the other day was by using this youtube-dl command:
` youtube-dl -f 95 -g https://www.youtube.com/watch\?v\=21qNxnCS8WU `
For non live stream video it doesn't give .m3u8 file.

**Comment by Rizary** [comment]
Based on the documentation, `/sync` endpoint is to sync live stream to 0Storage. Or is there any other use case that we need to update?

**Comment by chauchausoup** [comment]
I think there should be the validation layer before processing the url other wise the server may crash.

**Comment by cnlangzi** [comment]
it may work with name *.mp4 instead of *.m3u8. @chauchausoup

**Comment by chauchausoup** [comment]
> it may work with name *.mp4 instead of *.m3u8. @chauchausoup

@cnlangzi yeah at present if you pass normal youtube video then the 0proxy crashes and cant do further operation, so I was thinking of adding a validation layer at the proxy or as you mentioned to include .mp4 extension too which will enhance the experience.

**Comment by Rizary** [comment]
I get this: 
![image](https://user-images.githubusercontent.com/7221768/146203032-4f3bb0b6-ee38-480e-882e-967cecc11f6a.png)

when downloading normal youtube video. here is the result in the UI:
![image](https://user-images.githubusercontent.com/7221768/146203124-4c0f1ec4-16a1-454f-b26b-80e3d53be75c.png)

I don't see any crash in it.

**Comment by Rizary** [comment]
Here is the command I used:
```
$ curl -X POST -d 'allocation=6446202ee6f6a0bd5caec6ce88f42656173044af534cacfeabbef021f267e6b6' -d 'client_json={"client_id":"11a0da2da4e6735acc1e7a5477ab9bef96818f9173fc7975458d1a11cb457dba","client_key":"38dfe9cff6d5fee598f70aea8a996c549250dd3ccf297a24c2b7334a92f25701bf6bcb8e946c3008838f97c2f0d18b631f86b9858eb8d00e4f22f46782044401","keys":[{"public_key":"38dfe9cff6d5fee598f70aea8a996c549250dd3ccf297a24c2b7334a92f25701bf6bcb8e946c3008838f97c2f0d18b631f86b9858eb8d00e4f22f46782044401","private_key":"07ad025852ec2ee700fc650f396a5f34e527b9b3204c59cd1633fab71e102417"}],"mnemonics":"cook toilet adult sunset cry finger differ dream bring soul guilt luxury fence achieve must cabbage ostrich wonder science exchange rice small refuse april","version":"1.0","date_created":"53810-05-04 03:04:20"}' -d 'encrypt=false' -d 'feed=https://www.youtube.com/watch?v=4cJpiOPKH14' -d 'downloader_args=-f best' -d 'ffmpeg_args=' -d 'delay=5' -d 'remote_path=/sync/t3' -d 'file_name=video.m3u8' http://localhost:9082/sync
```

**Comment by chauchausoup** [comment]
Yes its working on the video you mentioned. But please check this video,
https://www.youtube.com/watch?v=vDHtypVwbHQ 
This is the one I checked the proxy/sync with and its crashing proxy container.
